### DISCUSSION TITLE
Test unitaire pour les scores avec 3 joueurs mais s'attend a en avoir 4

### DISCUSSION THREAD

**Discussion by doriangi** [web-classroom/web-classroom.github.io#11] [Q&A : Pratique / Labos] [answered]
Bonjour, 

Dans ce test unitaire :
![image](https://github.com/web-classroom/web-classroom.github.io/assets/71764824/b1d1fcb7-6af9-45b9-ac0d-e64f7b9c8c35)
Le game définit 3 joueurs (1, 2 et 4) et le test va ensuite vérifier que les scores des joueurs soient corrects. Mais on test aussi pour le joueur 3 qui n'est pas définit dans la liste des joueurs de la map. Est-ce que dans notre fonction `getTotalScores()` on est sensé déduire qu'il doit avoir un joueur 3 étant donné qu'on a un joueur 4, et devoir lui mettre un score de -1 ?

**Comment by OlivierLmr** [comment]
Bonjour,
Dans ce test, le joueur 3 ne fait effectivement pas partie des joueurs actifs, mais il existe une pièce lui appartenant sur la game-map. On suppose donc effectivement qu'un joueur qui a quitté la partie n'a plus de score lié au nombre de lignes qu'il ou elle a réussi à clear. Le -1 vient donc du nombre de blocs de sa couleur sur la carte, dans ce cas 1.

**Comment by OlivierLmr** [comment, marked as answer]
J'ai mis en ligne des précisions dans l'énoncé au sujet du comportement à suivre lorsqu'un player quitte la partie
